pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,490 @@
+
+    display_text_no_symbol = Instrument.setting(
+        "D3%s",
+        """Displays up to 12 upper-case ASCII characters on the display and disables all symbols on the display.
+        """,
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    measure_ACI = Instrument.measurement(
+        MODES["ACI"],
+        """Return the measured value for AC current
+        """,
+        )
+
+    measure_ACV = Instrument.measurement(
+        MODES["ACV"],
+        """Return the measured value for AC Voltage
+        """,
+        )
+
+    measure_DCI = Instrument.measurement(
+        MODES["DCI"],
+        """Return the measured value for DC current
+        """,
+        )
+
+    measure_DCV = Instrument.measurement(
+        MODES["DCV"],
+        """Return the measured value for DC Voltage
+        """,
+        )
+
+    measure_R2W = Instrument.measurement(
+        MODES["R2W"],
+        """Return the measured value for 2-wire resistance
+        """,
+        )
+
+    measure_R4W = Instrument.measurement(
+        MODES["R4W"],
+        """Return the measured value for 4-wire resistance
+        """,
+        )
+
+    measure_Rext = Instrument.measurement(
+        MODES["Rext"],
+        """Return the measured value for extended resistance mode (>30M, 2-wire) resistance
+        """,
+        )
+
+    mode = Instrument.control(
+        "B",
+        "%s",
+        """Return current selected measurment mode, this propery can be set.
+        Allowed values are ACI,ACV,DCI,DCV,R2W,R4W,Rext
+        """,
+        get_process = (lambda x: HP3478A.decode_mode(x)),
+        set_process = (lambda x: HP3478A.MODES[x]),
+        validator = strict_discrete_set,
+        values = MODES,
+        )
+
+    range = Instrument.measurement(
+        "B",
+        """Returns the current measurment range, this property can be set.
+        Valid settings are 3*powers of ten (e.g 0.3,3,30)"
+        for all valid ranges look at HP3478A.RANGES structure
+        """,
+        get_process = lambda x: HP3478A.decode_range(x),
+        )
+
+    @range.setter
+    def range(self,value):
+        cur_mode = self.mode
+        set_range = strict_discrete_set(value, self.RANGES[cur_mode])
+        set_range = self.RANGES[cur_mode][value]
+        self.write(set_range)
+
+    resolution = Instrument.control(
+        "B",
+        "N%d",
+        """Return current selected resolution, this property can be set.
+        Allowed values are 3,4 or 5
+        """,
+        get_process = (lambda x: 6-HP3478A.decode_status(x,"digits")),
+        validator = strict_discrete_set,
+        values = [3,4,5],
+        )
+
+    status = Instrument.measurement(
+        "B",
+        """Checks the status registers
+        """,
+        get_process = (lambda x: HP3478A.decode_status(x)),
+        )
+
+    SRQ_mask = Instrument.control(
+        "B",
+        "M%o",
+        """Return current SRQ mask, this property can be set,
+
+        bit assigment for SQR:
```

Comment:
"SRQ", I think.

Reply:
And decoding that for the user could also be dealt with with an IntFlag enum.

Reply:
Thanks, 
spelling fixed
and IntFlag enum added for SRQ mask decoding